### PR TITLE
Polyglot

### PR DESCRIPTION
**Pull request by craigjbass**

### CODE REVIEW

File: posts/polyglot.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+# Polyglot disciplines
+
+## Problems
+
+One of the biggest problems with switching between programming languages is the context switching. Remembering what operator is used to perform a particular language essential function e.g. `;` and `.` for statement termination or `.` or `->` for object member access.
```
Comment (adrianclay) [inline posts/polyglot.md]: Jumping straight into the problems feels a bit odd.  Could do with some sort of context/intro

File: posts/polyglot.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+Remembering all the differences between languages is hard work. Being a "specialist" in a language requires serious dedication, internalising knowledge about every corner of a language requires a lot of practical experience.
+
+So how do I remember all of this? The secret is: **I don't**. 
+
+Here, I'll tell you how I get by without being a specialist.
+
+### Be familiar with the paradigms
+
+It's important to learn new languages. The biggest piece of advice to those looking to become a polyglot is to pick languages that are vastly different to what you currently find comfortable.
```
Comment (adrianclay) [inline posts/polyglot.md]: > It's important to learn new languages

Citation needed

File: posts/polyglot.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+## My polyglot toolbox
+
+Remembering all the differences between languages is hard work. Being a "specialist" in a language requires serious dedication, internalising knowledge about every corner of a language requires a lot of practical experience.
+
+So how do I remember all of this? The secret is: **I don't**. 
+
+Here, I'll tell you how I get by without being a specialist.
+
+### Be familiar with the paradigms
```
Comment (adrianclay) [inline posts/polyglot.md]: Does this title fit?

File: posts/polyglot.md (hunk 2 above)
Comment (craigjbass) [inline posts/polyglot.md]: If you're trying to be a successful polyglot. Come on lol.

File: posts/polyglot.md (hunk 3 above)
Comment (craigjbass) [inline posts/polyglot.md]: Yeah maybe not

File: posts/polyglot.md (hunk 2 above)
Comment (adrianclay) [inline posts/polyglot.md]: Haha, I just read a whole section on all the problems.  You've anchored my view to be negative.

File: posts/polyglot.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+I have found that meetups e.g. London Code Dojo or London Software Craftsmanship are full of individuals who can provide mentoring over a kata. The subtle transfer of knowledge that takes place in less than 2 hours is substantial enough to be worth your time.
+
+### TDD
+
+The discipline with the biggest impact on my polyglot workflow is TDD. When I say TDD, I mean full on - Disciplined TDD. 
+
+TDD is a strict discipline where a programmer is not able to write a single line of production code without writing a failing test. Given the tests must be written, and they must be written first when TDD'ing a programmer is also only allowed to write the simplest assertion to produce a failing test. When a programmer has a failing test, they are also only allowed to write the simplest production code to make that test pass. What this results in in practice is that a TDD'ing programmer's test suite assertions have to be watertight.
```
Comment (craigjbass) [inline posts/polyglot.md]: "What this yields in practice is a test suite composed of watertight assertions."

File: posts/polyglot.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+### Defer Technology Decisions
+
+As an agilist one thing I want to be delivering is a continuous stream of value. One thing that I find to be of high friction is new technologies. 
+
+The most common decision that I would suggest deferring is the answer to this one question: "What is the database schema?". Some Engineers will gasp at this point, but hold back any disbelief in this for a moment. When decisions like what the database schema is and when you architect your code to be truly agnostic of a particular web framework, something interesting happens, you can worry about what your customer needs, without learning anything but a test framework and the language at hand. You can worry about learning how to solve those details later. Not only that, but you'll be more informed to make the right decision about the framework or the database as you'll have more knowledge about the shape of the solution. Maybe, a lightweight database is enough (or even flat files!), or maybe Sinatra is sufficient to get the job done.
```
Comment (craigjbass) [inline posts/polyglot.md]: "...particular web framework are deferred..."

**Comment by sowasred2012** [comment]
LGTM!

**Pull request merged**